src/rgw/setup/rgw.py

```diff
@@ -677,20 +676,28 @@ def _verify_backend_store_value(conf: MappedConf):
                 f' currently configured one is {backend_store}')
 
     @staticmethod
-    def _update_svc_config(conf: MappedConf, client_section: str, config_key_mapping: list):
+    def _update_svc_config(conf: MappedConf, client_section: str, config_key_mapping: dict):
         """Update config properties from confstore to rgw config file."""
         svc_config_file = Rgw._get_rgw_config_path(conf)
         Rgw._load_rgw_config(Rgw._conf_idx, f'ini://{svc_config_file}')
-        Log.info(f'adding paramters to {client_section} in {svc_config_file}')
+        Log.info(f'updating paramters to {client_section} in {svc_config_file}')
 
-        # e.g config_key_mapping = [[confstore_key1, actual_svc_config_key1, default_value1],
-        # [confstore_key2, actual_svc_config_key2, default_valu2], ..]
-        for confstore_key, config_key, default_value in config_key_mapping:
-            # fetch actual value of parameter from confstore.
-            # if config key/value is missing in confstore then use default value mentioned in const.py
-            config_value = Rgw._get_cortx_conf(conf, confstore_key, default_value)
-            Log.info(f'Setting config key :{config_key} with value:{config_value} at {client_section} section')
-            Conf.set(Rgw._conf_idx, f'{client_section}>{config_key}', str(config_value))
+        for config_key, confstore_key in config_key_mapping.items():
+            if confstore_key is None:
+                Log.info(f'config key:{config_key} not found in rgw key mapping.\
+                    hence using default value for this key')
+                continue
+            else:
+                # fetch actual value of parameter from confstore.
+                # if config key/value is missing in confstore then use default value mentioned in config file.
+                config_value = conf.get(confstore_key)
+                if config_value is None:
+                    Log.info(f'GConfig entry is missing for config key :{config_key}.\
+                        hence using default value specified in config file.')
+                    continue
+                else :
+                    Log.info(f'Setting config key :{config_key} with value:{config_value} at {client_section} section')
```

Comment:
put code with higher probability first in execution flow (code under 'else') and error handling in else case (right now under 'if' )

Reply:
done